src/conversion.rs

```diff
@@ -178,6 +179,163 @@ pub trait IntoPy<T>: Sized {
     }
 }
 
+/// Owned or borrowed gil-bound Python smart pointer
+pub trait AnyBound<'py, T> {
```

Comment:
Yes, I think we should seal this. I haven't done that already because I wasn't sure where we want that trait to live. Should it stay next to `IntoPyObject`, next to `Bound` or in its own module...

Reply:
I would put it next to `Bound` perhaps (i.e. in `instance.rs`, so that its only reachable path will be via `lib.rs` (and maybe the prelude).

Reply:
I moved it, reexported it from `lib.rs` and added the sealing. We _probably_ want it in the prelude, but I left that for later when we captured a bit more experience.